app/models/user.rb

```diff
@@ -133,4 +133,17 @@ def downcase_keys
   def strip_whitespace
     # no-op
   end
+
+  # Override Devise because we want to include the SAML request id in the
+  # confirmation instructions email
+  def send_confirmation_instructions
+    # no-op
+  end
+
+  def send_custom_confirmation_instructions(id = nil)
+    generate_confirmation_token! unless @raw_confirmation_token
```

Comment:
Yes, I copied Devise's `send_confirmation_instructions` with the only modification being the ability to pass in the request_id to the mailer. I'll add a comment.